include/wx/unix/glegl.h

```diff
@@ -0,0 +1,172 @@
+#include <GL/gl.h>
+
+// This is to avoid including EGL headers here to pollute namespace
+typedef void *EGLDisplay;
+typedef void *EGLConfig;
+typedef void *EGLSurface;
+typedef void *EGLContext;
+
+class wxGLContextAttrs;
+class wxGLAttributes;
+
+// ----------------------------------------------------------------------------
+// wxGLContext
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_GL wxGLContext : public wxGLContextBase
+{
+public:
+    wxGLContext(wxGLCanvas *win,
+                const wxGLContext *other = NULL,
+                const wxGLContextAttrs *ctxAttrs = NULL);
+    virtual ~wxGLContext();
+
+    virtual bool SetCurrent(const wxGLCanvas& win) const wxOVERRIDE;
+
+private:
+    EGLContext m_glContext;
+
+    wxDECLARE_CLASS(wxGLContext);
+};
+
+// ----------------------------------------------------------------------------
+// wxGLCanvasEGL
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_GL wxGLCanvasEGL : public wxGLCanvasBase
+{
+public:
+    // initialization and dtor
+    // -----------------------
+
+    // default ctor doesn't do anything, InitConfig() must be called
+    wxGLCanvasEGL();
+
+    // initializes EGLConfig corresponding to the given attributes
+    bool InitVisual(const wxGLAttributes& dispAttrs);
+
+    // creates EGLSurface
+    bool CreateSurface();
+
+    virtual ~wxGLCanvasEGL();
+
+
+    // implement wxGLCanvasBase methods
+    // --------------------------------
+
+    virtual bool SwapBuffers() wxOVERRIDE;
+
+
+    // X11-specific methods
+    // --------------------
+
+    // get the X11 handle of this window
+    virtual unsigned long GetXWindow() const = 0;
+
+
+    // override some wxWindow methods
+    // ------------------------------
+
+    // return true only if the window is realized: OpenGL context can't be
+    // created until we are
+    virtual bool IsShownOnScreen() const wxOVERRIDE;
+
+
+    // implementation only from now on
+    // -------------------------------
+
+    // get the EGLConfig we use
+    EGLConfig *GetEGLConfig() const { return m_config; }
+    EGLDisplay GetEGLDisplay() const { return m_display; }
+    EGLSurface GetEGLSurface() const { return m_surface; }
+
+    static EGLDisplay GetDisplay();
+
+    // initialize the global default GL config, return false if matching config
+    // not found
+    static bool InitDefaultConfig(const int *attribList);
+
+    // get the default EGL Config (may be NULL, shouldn't be freed by caller)
+    static EGLConfig *GetDefaultConfig() { return ms_glEGLConfig; }
+
+    // free the global GL visual, called by wxGLApp
+    static void FreeDefaultConfig();
+
+    // initializes EGLConfig
+    //
+    // returns false if EGLConfig couldn't be initialized, otherwise caller
+    // is responsible for freeing the pointers
+    static bool InitConfig(const wxGLAttributes& dispAttrs, EGLConfig **pEGLC);
+
+    bool m_ready_to_draw;
```

Comment:
Minor, but it would be better to use the usual naming convention:
```suggestion
    bool m_readyToDraw;
```